File: testrobot_templates/cpp/src/robot.cpp
#include <robot.hpp>

#include <arpirobot/core/log/Logger.hpp>
#include <arpirobot/core/action/ActionManager.hpp>
#include <arpirobot/core/network/NetworkTable.hpp>

#include <string>

using namespace arpirobot;


void Robot::robotStarted(){
    arduino.addDevice(&imu);
    arduino.begin();
    imu.calibrate(10);

    NetworkTable::set("P", "1.0");
    NetworkTable::set("I", "0.0");
    NetworkTable::set("D", "0.0");

    flmotor.setInverted(true);
    frmotor.setInverted(true);

    flmotor.setBrakeMode(true);
    frmotor.setBrakeMode(true);
    rrmotor.setBrakeMode(true);
    rlmotor.setBrakeMode(true);

    myTestAction.setProcessPeriodMs(20);
}

void Robot::robotEnabled(){
    ActionManager::startAction(&myTestAction);
}

void Robot::robotDisabled(){
    ActionManager::stopAction(&myTestAction);
}

void Robot::enabledPeriodic(){
    
}

void Robot::disabledPeriodic(){

}

void Robot::periodic(){
    NetworkTable::set("Gyro", std::to_string(imu.getGyroZ()));
    // Do not remove this line or some devices will be disabled.
    feedWatchdog();
}

File: testrobot_templates/py/robot.py

from arpirobot.core.action import ActionManager
from arpirobot.core.log import Logger
from arpirobot.core.network import NetworkTable
from arpirobot.core.robot import BaseRobot

class Robot(BaseRobot):
    def __init__(self):
        super().__init__()

    def robot_started(self):
        pass

    def robot_enabled(self):
        pass

    def robot_disabled(self):
        pass

    def periodic(self):
        # Always make sure to feed the watchdog to make sure devices don't get disabled
        self.feed_watchdog()
    
    def enabled_periodic(self):
        pass

    def disabled_periodic(self):
        pass
